### DISCUSSION TITLE
Getting separate force values

### DISCUSSION THREAD

**Discussion by mojtabaAmiraslanpour** [UvaCsl/HemoCell#117] [General discussion]
Hi, I need access to separate force values for each cell particle. But I just found out that the forces are combined by default:

```  serializeValues_t sv;

  hemo::Array<T,3> force_total;
  plint tag;
  #ifdef INTERIOR_VISCOSITY
  hemo::Array<T,3> normalDirection;
  //Is vector, optimize with hemo::Array possible
  std::vector<hemo::Array<plint, 3>> kernelCoordinates;
  #endif

  std::vector<plb::Cell<T,DESCRIPTOR>*> kernelLocations;
  std::vector<T>         kernelWeights;

  hemo::Array<T,3> *force_volume = &sv.force;
  hemo::Array<T,3> *force_bending = &sv.force;
  hemo::Array<T,3> *force_link = &sv.force;
  hemo::Array<T,3> *force_area = &sv.force; //Default to pointing to force, if the output is desired, it can be stored separately
  hemo::Array<T,3> *force_visc = &sv.force;
  hemo::Array<T,3> *force_inner_link = &sv.force;
```

How can I get separate forces? I am also confused about how HDF5 files contain separate forces.

Thanks,
Mojtaba

**Comment by gzavo** [comment]
Indeed, the forces are separated specifically at the iteration when you save. As many other complexities in HemoCell, this has to do with performance. Check for the uses of the following function:
https://github.com/UvaCsl/HemoCell/blob/d91698721f7a06166cb6ac8d22fa2f2f4baa4ed7/core/hemoCellParticleField.cpp#L590

**Comment by mojtabaAmiraslanpour** [comment]
Thank you! I see how it is defined now.

For test purposes I made it happen in the `writeOutput()` function right after `applyConstitutiveModel`.
